libs/ledger/src/chaincode/chain_code_cache.cpp

```diff
@@ -30,30 +30,66 @@
 namespace fetch {
 namespace ledger {
 
-ChainCodeCache::ContractPtr ChainCodeCache::Lookup(Identifier const &contract_id,
-                                                   StorageInterface &storage)
+namespace {
+
+using byte_array::ConstByteArray;
+
+template <std::size_t MIN_LENGTH, std::size_t MAX_LENGTH>
+bool IsBase58(ConstByteArray const &value)
+{
+  char const *buffer = value.char_pointer();
+
+  if (!(value.size() >= MIN_LENGTH && value.size() <= MAX_LENGTH))
+  {
+    return false;
+  }
+
+  for (std::size_t i = 0; i < value.size(); ++i, ++buffer)
+  {
+    // 1-9A-HJ-NP-Za-km-z
+    bool const valid =
+        ((('1' <= *buffer) && ('9' >= *buffer)) || (('A' <= *buffer) && ('H' >= *buffer)) ||
+         (('J' <= *buffer) && ('N' >= *buffer)) || (('P' <= *buffer) && ('Z' >= *buffer)) ||
+         (('a' <= *buffer) && ('k' >= *buffer)) || (('m' <= *buffer) && ('z' >= *buffer)));
+
+    if (!valid)
+    {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+bool IsIdentity(ConstByteArray const &value)
+{
+  return IsBase58<48, 50>(value);
+}
+
+}  // namespace
+
+ChainCodeCache::ContractPtr ChainCodeCache::Lookup(ConstByteArray const &contract_id,
+                                                   StorageInterface &    storage)
 {
   // attempt to locate the contract in the cache
   ContractPtr contract = FindInCache(contract_id);
 
   // if this fails create the contract
   if (!contract)
   {
-    if (contract_id.type() == Identifier::Type::SMART_OR_SYNERGETIC_CONTRACT)
+    if (IsIdentity(contract_id))
```

Comment:
I would recommend using the `chain::Address::Parse` function here. This will do a better job with the parse and will also validate the addresses built in checksum.